app/src/main/java/net/squanchy/tweets/service/TwitterService.java

```diff
@@ -0,0 +1,118 @@
+package net.squanchy.tweets.service;
+
+import com.twitter.sdk.android.core.Callback;
+import com.twitter.sdk.android.core.Result;
+import com.twitter.sdk.android.core.TwitterException;
+import com.twitter.sdk.android.core.models.Tweet;
+import com.twitter.sdk.android.tweetui.Timeline;
+import com.twitter.sdk.android.tweetui.TimelineResult;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.squanchy.tweets.view.TimelineStateHolder;
+
+public class TwitterService {
+
+    private final Timeline<Tweet> timeline;
+    private final TimelineStateHolder timelineStateHolder;
+    private List<Tweet> itemList;
+
+    public TwitterService(Timeline<Tweet> timeline) {
+        this.timeline = timeline;
+        this.timelineStateHolder = new TimelineStateHolder();
+
+        itemList = new ArrayList<>();
+    }
+
+    public void refresh(Callback<TimelineResult<Tweet>> developerCb) {
+        timelineStateHolder.resetCursors();
+        load(timelineStateHolder.positionForNext(), new RefreshCallback(developerCb, timelineStateHolder));
+    }
+
+    public void previous(Callback<TimelineResult<Tweet>> developerCb) {
+        loadPrevious(timelineStateHolder.positionForPrevious(), new PreviousCallback(developerCb, timelineStateHolder));
+    }
+
+    public int size() {
+        return itemList.size();
+    }
+
+    public Tweet itemAt(int position) {
+        return itemList.get(position);
+    }
+
+    private void load(Long minPosition, Callback<TimelineResult<Tweet>> cb) {
+        if (timelineStateHolder.startTimelineRequest()) {
+            timeline.next(minPosition, cb);
```

Comment:
what happens if `minPosition` is null here? (which can be in case `timelineStateHolder` returns a null there)